### PR TITLE
[Enhancement] Support includes/excludes with text fields for bulk filtering.

### PR DESCRIPTION
**Pull request by heathdutton**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | Yes
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | #5924
| BC breaks? | No
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
See #5924

#### Steps to test this PR:
1. Create a new segment "Test includes/excludes segment"
2. Filter this segment by "Zip Code" with "includes" and put in several zip codes that you know some of your contacts have as a string. Any listing format should suffice. Example strings below.
3. Run `php console mautic:segments:update` or wait for your cron to execute.
4. The new segment should have the contacts that are within those zipcodes.

This should work with any text field, I'm just using ZipCodes as a real-world example.

Example: Random zips without spaces from the demo data. As copied by Numbers. Should auto-delimit by space.
```
27401 2870 30458 3263 3312 3342 33762 347 4357 451
```
Example: Some UK zips as produced by Excel when copying. Should auto-delimit by space using quote as the enclosure.
```
"GL15 1XD" "HS6 2YL" "IV28 9UH" "J0K 1X0" "J0R 1C0" "J4H 1M3" "K1H 7Z1" "L8G 2V1" "LA6 6ZT" "N9A 1H9" "PA31 8WG" "SY10 6ND" "SY22 2AD" "TA11 7UU" "TA18 7XS" "V0E 2P0" "V0K 1A0" "V6B 3K9" "ZE2 4FH"
```
Example: All zipcodes in demo data, copied from a CSV.
```
 "07020",10965,11201,1220,1512,1544,17036,2102,220,2337,27401,2870,30458,3263,3312,3342,33762,347,4357,451,4671,4860,49077,49221,50313,505,5185,57401,6204,6312,6475,705,7435,87112,"CH5 9LB","GL15 1XD","HS6 2YL","IV28 9UH","J0K 1X0","J0R 1C0","J4H 1M3","K1H 7Z1","L8G 2V1","LA6 6ZT","N9A 1H9","PA31 8WG","SY10 6ND","SY22 2AD","TA11 7UU","TA18 7XS","V0E 2P0","V0K 1A0","V6B 3K9","ZE2 4FH"
```

### CODE REVIEW

**Comment by npracht** [comment]
Label: Ready to test

**Comment by heathdutton** [comment]
@npracht I love the suggestion. I'll backlog a task for that enhancement. This PR was just to get something functional.

I'll have to test it a bit to see if the tag renderer can handle ~10k tags in the DOM without crashing IE :)
Also, I feel the field needs to auto-expand a bit to be able to see more entries when you have a high number of them in there (especially within campaign filters). Something like a textarea with a maximum height of 10 rows and a vertical scroll... (I'll have to play with it).

**Comment by heathdutton** [comment]
I'm going to remove the compiled JS, and leave that out of future PRs. That way the patch process is clean (can apply multiple at once with composer-patches).

File: app/bundles/LeadBundle/Entity/LeadListRepository.php [hunk 1]
```diff
@@ -1677,11 +1677,44 @@ public function getListFilterExpr($filters, &$parameters, QueryBuilder $q, $isNo
 
                         case 'in':
                         case 'notIn':
+                            if (is_string($details['filter'])) {
+                                // Expound the string to an array based on common string array notations.
+                                foreach ([',', ';', "\t", ' '] as $delimiter) {
```
Comment (escopecz) [inline app/bundles/LeadBundle/Entity/LeadListRepository.php]: I don't like the idea of spaces as delimiters. Users will fill in value like `New York, Boston`. Building arrays from a text value is used elsewhere in Mautic too. For for example defining multiple To/CC/BCC email addresses. Only commas are used as separators in all those places. Do you think you could remove the other delimiters?

Then you could use simpler parsing by using [str_getcsv](http://php.net/manual/en/function.str-getcsv.php) function.

**Comment by npracht** [comment]
Hi @heathdutton what do we do with it ?

File: app/bundles/LeadBundle/Entity/LeadListRepository.php (hunk 1 above)
Comment (heathdutton) [reply app/bundles/LeadBundle/Entity/LeadListRepository.php]: These are "possible" delimiters. So it is compatible with str_getcsv, spaces only being the last possible option. I was trying to support all the different kinds of spreadsheet apps here... 
That being said this should be done by JS upon pasting in values, not in the PHP backend. That way there would be no question how the separate values would be treated.

I think I'll refactor this PR today.

**Comment by alanhartless** [comment]
@heathdutton I'm closing this as I'm trying to clean up lingering PRs. You mentioned refactoring it. Can you re-submit once that's done? Thanks!

**Pull request closed without merging**